make-rules/shared-macros.mk

```diff
@@ -345,6 +348,15 @@ export PARFAIT_NATIVESUNCXX=$(SPRO_VROOT)/bin/CC
 export PARFAIT_NATIVEGCC=$(GCC_ROOT)/bin/gcc
 export PARFAIT_NATIVEGXX=$(GCC_ROOT)/bin/g++
 
+# This should evaluate to empty string or a pathname like /usr/bin/ccache
+# depending on current PATH value and "which" implementation.
+# The assignment via ":=" is important, to only do this once in a Makefile,
+# and not on every reference to the value as "=" assignment would result in.
+# Review `man ccache` for optional configuration tuning, like cache size etc.
+# For production builds or suspected errors you can disable this feature by
+# setting CCACHE_DISABLE=true (as an environment or makefile variable value).
+export CCACHE := $(shell if test -n "$(CCACHE)" ; then echo "$(CCACHE)"; else test x"$${CCACHE_DISABLE-}" = xtrue -o x"$(CCACHE_DISABLE)" = xtrue || for F in "$$CCACHE" `which ccache 2>/dev/null | egrep '^/'` /usr/bin/ccache; do test -x "$$F" && echo "$$F" && echo "USING CCACHE FOR OI-USERLAND: $$F" >&2 && break; done; fi)
```

Comment:
Please, make this more readable... Perhaps, we can rewrite it using less shell and more make expressions, or split it in several steps. At least, split it in several lines...